src/articles/components/ArticleView.tsx

```diff
@@ -0,0 +1,27 @@
+import React, { useEffect, useState } from 'react';
+import { usePrefetch } from '../../common/hooks/usePrefetch';
+import { PrefetchKey } from '../../common/utils/PrefetchState';
+import { getArticle } from '../api';
+import { mockArticle } from '../models/Article';
+
+export interface IProps {
+  articleId: number;
+}
+
+export const ArticleView = ({ articleId }: IProps) => {
+  const prefetchArticle = usePrefetch(PrefetchKey.ARTICLE_SINGLE, async () => await getArticle(articleId));
+  const defaultArticle = prefetchArticle && prefetchArticle.id === articleId ? prefetchArticle : mockArticle;
+
+  const [article, setArticle] = useState(defaultArticle);
+
+  useEffect(() => {
+    const fetchArticle = async () => {
+      const newArticle = await getArticle(articleId);
+
+      setArticle(newArticle);
+    };
+    fetchArticle();
+  }, []);
```

Comment:
Should probably add `articleId` to the dependency array to make sure the article is fetched when you go from one article to another.

Reply:
Good catch! Fixed it 😄